server/src/main/java/org/apache/druid/client/DirectDruidClient.java

```diff
@@ -465,38 +467,31 @@ public void onFailure(Throwable t)
             {
               openConnections.getAndDecrement();
               if (future.isCancelled()) {
-                // forward the cancellation to underlying queriable node
-                try {
-                  StatusResponseHolder res = httpClient.go(
-                      new Request(
-                          HttpMethod.DELETE,
-                          new URL(cancelUrl)
-                      ).setContent(objectMapper.writeValueAsBytes(query))
-                       .setHeader(
-                           HttpHeaders.Names.CONTENT_TYPE,
-                           isSmile
-                           ? SmileMediaTypes.APPLICATION_JACKSON_SMILE
-                           : MediaType.APPLICATION_JSON
-                       ),
-                      StatusResponseHandler.getInstance(),
-                      Duration.standardSeconds(1)
-                  ).get(1, TimeUnit.SECONDS);
-
-                  if (res.getStatus().getCode() >= 500) {
-                    throw new RE(
-                        "Error cancelling query[%s]: queriable node returned status[%d] [%s].",
-                        query,
-                        res.getStatus().getCode(),
-                        res.getStatus().getReasonPhrase()
-                    );
+                Runnable cancelRunnable = () -> {
+                  try {
+                    httpClient.go(
+                        new Request(HttpMethod.DELETE, new URL(cancelUrl))
+                        .setContent(objectMapper.writeValueAsBytes(query))
+                        .setHeader(HttpHeaders.Names.CONTENT_TYPE, isSmile ? SmileMediaTypes.APPLICATION_JACKSON_SMILE : MediaType.APPLICATION_JSON),
+                        StatusResponseHandler.getInstance(),
+                        Duration.standardSeconds(1));
                   }
-                }
-                catch (IOException | ExecutionException | InterruptedException | TimeoutException e) {
-                  throw new RuntimeException(e);
-                }
+                  catch (IOException e) {
+                    log.error("Error cancelling query[%s]: ", query);
+                  }
+                };
+
+                Future<?> responseFuture = queryCancellationExecutor.submit(cancelRunnable);
+                Runnable checkRunnable = () -> {
+                  if (!responseFuture.isDone()) {
+                    log.error("Error cancelling query[%s]: ", query);
+                  }
+                };
+                queryCancellationExecutor.schedule(checkRunnable, 1, TimeUnit.SECONDS);
```

Comment:
I would also now increase the check timeout to 5 seconds because it's now async. 1 second feels like it may be a little too short and it might create false positives. E. g. there may be a GC pause on Historical, easily 1-2 second long. We don't want to consider the cancellation request failed just because it happens during such GC pause.